internal/command/cmd.go

```diff
@@ -119,6 +146,34 @@ func (c *Cmd) buildCmd() (*exec.Cmd, context.CancelFunc) {
 	return cmd, cancelFun
 }
 
+func (c *Cmd) sanitisedCmd() string {
+	if c.Command == "" {
+		return c.Command
+	}
+
+	// split the command string on " ".
+	cmdStrings := strings.Split(c.Command, " ")
```

Comment:
I am not a fan of this approach. We are assuming that `kubectl`, or any other command we would want to sanitise in future would have to be first. I think we could do better.

What I had in mind originally would be to call `SanitiseXXX()` on every command, regardless if the command is `kubectl`, `terraform` or whatnot. If the secret regex would match, we would replace it, if not, we would just return the original command. Maybe we will be calling `SanitiseXXX()` unnecessarily, however, we would be sure that no secret would be leaked in the logs.

Reply:
> We are assuming that `kubectl`, or any other command we would want to sanitise in future would have to be first.

well..as long as this assumed behaviour is covered by tests..which it is.

> What I had in mind originally would be to call `SanitiseXXX()` on every command, regardless if the command is `kubectl`, `terraform` or whatnot. If the secret regex would match, we would replace it, if not, we would just return the original command. Maybe we will be calling `SanitiseXXX()` unnecessarily, however, we would be sure that no secret would be leaked in the logs.

I see. this approach has its own risks, however, for instance mangling the logs potentially by replacing inputs that were not meant to be passed in..
I can give it a rework, though.

Reply:
the way it is now we know what we're after and where and apply precise methods to achieve it.
with generic solutions it will definitely be easier to miss stuff.

Reply:
so to clarify, did you mean to run *any* command through all available sanitisation functions? I originaly got the idea that we were going to selectively choose what to sanitise and how.. @MiroslavRepka

Reply:
>did you mean to run any command

yes, I did. We could selectively choose which one to sanitise, however, the algorithm you have for it now is not enough. We can add commands like `kustomize build | kubectl --kubeconfig ..` later, which would not be caught by your solution. Instead of processing the commands to select which functions to call for sanitising, we can just sanitise it right away, even if we end up with the same command